README-VIS.md

```diff
@@ -0,0 +1,104 @@
+document quickly once over to get an overview, and then more slowly to get the details.
+
+## Building the image
+
+If you haven't worked with containers before, read this
+[introduction to Docker](https://docs.docker.com/get-started/), and try the
+examples in this [introcution to containers](https://docs.docker.com/get-started/part2/).
+The container ecology is complicated, but you only need to focus on one part,
+building your container and pushing it to a repository,
+and hopefully you can base your work on one of our examples.
+
+From the introduction, you should have the sense that containers are built up
+from layers, and it is important to start with the right first layer.
+Any popular language or framework will have a base image you can begin from,
+listed at the top of you `Dockerfile`: after that you will add additional
+libraries you require, and finally the code for your particular project.
+Here are samples from some of our projects:
+
+- [Python](https://github.com/refinery-platform/heatmap-scatter-dash/blob/master/context/Dockerfile)
+- [R Shiny](https://github.com/refinery-platform/intervene-refinery-docker/blob/master/context/Dockerfile)
+- [JS](https://github.com/refinery-platform/lineup-refinery-docker/blob/master/context/Dockerfile)*
+
+The JS example is odd, because although the real code is JS, a small script to
+process the input was required, so the base image is Python.
+
+It is best for your `Dockerfile` to be under version control. You might include
+it in the repo for your project, and have Docker builds be
+triggered with each check in: This is what we've done in the Python and
+Shiny examples above. In other cases, because of build complexity, or because
+you're wrapping a third party tool, the real source code will be in a separate
+repo that you only reference: When that repo is updated, you'll need to bump
+a corresponding version number in your wrapper.
+
+## Inputs
+
+When your container starts up, an `INPUT_JSON_URL` environment variable will be available.
+(It is good practice and can make development easier to also check `INPUT_JSON`.)
+Its content will look something like [this](https://github.com/refinery-platform/docker_igv_js/blob/master/input_fixtures/good/input.json).
+- Input files are provided as a list of URLs under `file_relationships`.
+- Detailed metadata for each file in the dataset, not just your input, is provided
+in `node_info`.
+- User supplied strings or numbers are available in `parameters`.
+
+## Development
+
+To run your container locally, check out this project and add a description of your
+tool to `tools.py`, and start the demo with `python ./manage.py runserver`.
+The `image` field can reference images either on DockerHub, or in your local cache.
+
+## Gotchas
+
+There are a few things to be careful about when wrapping your tool.
+
+- **Port 80**: `django_docker_engine` assumes your tool will use port 80: If you're using
```

Comment:
80. If you......

Reply:
If you're using a different port, that needs to be communicated when you publish your took, see the last section below. ??